src/services/connection.service.ts

```diff
@@ -0,0 +1,28 @@
+import { ConnectionsEntity } from '@/entities';
+import { GetUserConnectionsResponseDto } from '@/dtos';
+import { pgDataSource } from '@/databases';
+
+class ConnectionService {
+  public async getUserConnections(userId: number): Promise<GetUserConnectionsResponseDto> {
+    const connections = await ConnectionsEntity.find({
+      select: {
+        sender: { id: true, firstName: true, lastName: true, profilePic: true },
+        receiver: { id: true, firstName: true, lastName: true, profilePic: true },
+        id: true,
+        connected: true,
+      },
+      relations: { sender: true, receiver: true },
+      where: [
+        { sender: { id: userId }, connected: true },
+        { receiver: { id: userId }, connected: true },
+      ],
+    });
+    const userConnections = connections.map(({ id, connected, sender, receiver }) => {
```

Comment:
Curious. What happens if the user has no connections. What does this return you?

Reply:
empty array